### PR TITLE
Allow PHP 8 & require JMS serializer 3

### PR DESCRIPTION
**Pull request by okdewit**
After extensively reading the upgrade guides, I can't find any direct compatibility issues with Eversign SDK. 
Might warrant further testing though 😃

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -22,8 +22,8 @@
         "sami/sami": "3.3.*"
     },
     "require": {
-        "php": "^7.1",
+        "php": "^7.1|^8.0",
         "guzzlehttp/guzzle": "^6.2.3|^7.0.0",
-        "jms/serializer": "^1.6.2"
+        "jms/serializer": "^3.12.0"
```
Comment (milanlatinovic) [inline composer.json]: Would it make sense to support both major versions here "^1.6.2|^3.12.0" 

I checked jms/serializer official page and git repository, I see that the support was until 2020.
However there are breaking changes between major versions and maybe some members of our eversign community can not update to 3.x easily.

**Comment by milanlatinovic** [comment]
@okdewit hi, thanks for the contribution once more :D 
I added one comment, otherwise seems good.

bdw, seems like we have one more request for supporting PHP 8
https://github.com/eversign/eversign-php-sdk/issues/43

**Comment by okdewit** [comment]
@milanlatinovic Where can I find the comment?

Let me know if there's anything I can do to help this get through :smiley:

**Comment by milanlatinovic** [comment]
I just added the comment :) I'm thinking if it makes sense to support both existing and 3.x version of jms/serializer

File: composer.json (hunk 1 above)
Comment (okdewit) [reply composer.json]: Sure! Updated. :+1:

**Comment by okdewit** [comment]
Updated! :+1:

**Comment by milanlatinovic** [comment]
Fantastic, I'll take some time and check it out locally, but I do not expect any issues.
Will let you know about it asap.

**Comment by milanlatinovic** [comment]
@okdewit I checked PHP 8 compatibility locally, run linting, did some tests, also checked it manually. I could not recognize any issues. I will approve it and release new version.

I also noticed that in big part of SDK we do not have strong typing (so we do support PHP 7.x and PHP 8.x but majority of the code is PHP 5.x syntax (weak typing with comments). I will make a followup MR to tackle this, if you see it, I would appreciate a review. :D 

In short, approved, thanks!

**Comment by milanlatinovic** [comment]
I made a new release https://github.com/eversign/eversign-php-sdk/releases/tag/1.21.0

**Pull request merged**